src/webview/git-import/gitImportLoader.ts

```diff
@@ -26,19 +25,32 @@ let panel: vscode.WebviewPanel;
 export class Command {
     @vsCommand('openshift.component.importFromGit')
     static async createComponent(event: any) {
-        const workspacePath = await selectWorkspaceFolder();
-        const appendedUri = Uri.joinPath(workspacePath, event.projectName);
-        //const wsFolderLength = workspace?.workspaceFolders?.length || 0;
+        let alreadyExist: boolean;
+        let workspacePath,appendedUri: vscode.Uri;
+        do {
+            alreadyExist = false;
+            workspacePath = await selectWorkspaceFolder();
+            appendedUri = vscode.Uri.joinPath(workspacePath, event.projectName);
+            if (isWorkspaceFolder(workspacePath) && fs.readdirSync(workspacePath.fsPath).length > 0) {
+                vscode.window.showErrorMessage(`Unable to create Component on Workspace Folder: ${workspacePath}, Please select another folder`);
+                alreadyExist = true;
+            } else if (fs.existsSync(appendedUri.fsPath) && fs.readdirSync(appendedUri.fsPath).length > 0) {
+                vscode.window.showErrorMessage(`Folder ${appendedUri.fsPath.substring(appendedUri.fsPath.lastIndexOf('\\') + 1)} already exist
+                    at the selected location: ${appendedUri.fsPath.substring(0, appendedUri.fsPath.lastIndexOf('\\'))}`);
+                alreadyExist = true;
+            }
+        } while (alreadyExist);
         panel.webview.postMessage({
             action: 'cloneStarted'
-        })
-        //workspace.updateWorkspaceFolders(wsFolderLength, 0, { uri: appendedUri });
-        workspace.updateWorkspaceFolders(workspace.workspaceFolders? workspace.workspaceFolders.length : 0 , null, { uri: appendedUri });
+        });
+        if (!isWorkspaceFolder(workspacePath)) {
+            vscode.workspace.updateWorkspaceFolders(vscode.workspace.workspaceFolders ? vscode.workspace.workspaceFolders.length : 0, null, { uri: appendedUri });
```

Comment:
Try to do that on empty workspace. As soon as you add folder to workspace VSCode will reload the window and code after this line is not guarantee to be executed.